source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -265,7 +288,10 @@ void AuthenticatorImpl::verifyKey() {
   if (set_payload_cb_ && !provider.payload_in_metadata().empty()) {
     set_payload_cb_(provider.payload_in_metadata(), jwt_->payload_pb_);
   }
-
+  if (provider_ && jwks_data_->getJwtProvider().enable_token_cache() && !use_cache_jwt_) {
+    ENVOY_LOG(debug, "JWT {} added in cache", curr_token_->token());
+    jwks_data_->getTokenCache().insert(curr_token_->token(), jwt_, 1);
```

Comment:
Just check the SimpleLRUCache code,  it is taking pointer and own the pointer.  So here you need to do

jwks_data_->getTokenCache().insert(curr_token_->token(), jwt_.release(), 1);

to pass the pointer over to the cache

Reply:
fixed, called release on owned_jwt_.